politeiawww/comments.go

```diff
@@ -147,3 +160,72 @@ func validateComment(c www.NewComment) error {
 	_, err := util.ConvertStringToken(c.Token)
 	return err
 }
+
+// updateResultsForCommentLike updates the comment total votes, the votes
+// results and the vote resultant action for the user
+//
+// This function must be called WITHOUT the mutex held
+func (b *backend) updateResultsForCommentLike(like www.LikeComment) (*www.Comment, error) {
+	b.Lock()
+	defer b.Unlock()
+	return b._updateResultsForCommentLike(like)
+}
+
+// _updateResultsForCommentLike updates the comment total votes, the votes
+// results and the vote resultant action for the user
+//
+// This function must be called WITH the mutex held
+func (b *backend) _updateResultsForCommentLike(like www.LikeComment) (*www.Comment, error) {
```

Comment:
This looks expensive, can we optimize this a bit?

Reply:
It is not an expensive call. Even though the code looks a bit complex, none of the operations run inside this function is costly. It grabs a value from the inventory, do some pretty basic math calculus and update that value in the inventory. So I think it is okay to make the entire function atomic.
What bits do you see as expensive for this function?

Reply:
From slack resolution:
```
fernandoabolafio [3:35 PM]
Yeah, I can optimize that
okay, so the last one:
https://github.com/decred/politeia/pull/610/files#diff-c4d289a7e5b50d4535278b93a68b671aR178
You say that this function `_updateResultsForCommentLike(like www.LikeComment) (*www.Comment, error)` looks expensive.
I don’t think it is an expensive call. Even though the code looks a bit complex, none of the operations run inside this function is costly. It grabs a value from the inventory, do some pretty basic math calculus and update that value in the inventory. So I think it is okay to make the entire function atomic. What’s your concern? (edited)

moo31337 [3:53 PM]
ok
i can deal with that
```